### ISSUE TITLE
Training: inconsistent tensor size

### ISSUE THREAD

**Issue by bamos** [cmusatyalab/openface#36]
```
training(master*)$ ./main.lua -data ~/openface/data/casia-facescrub/dlib-affine-224-split/ -imgDim 224 -batchSize 20 -nDonkeys -1

==> doing epoch on training data:
==> online epoch # 1

/home/bamos/torch/install/bin/luajit: inconsistent tensor size at /home/bamos/torch/pkg/torch/lib/TH/generic/THTensorCopy.c:7
stack traceback:
        [C]: at 0x7fecb5c9aa60
        [C]: in function '__newindex'
        /home/bamos/openface/training/train.lua:132: in function 'f2'
        /home/bamos/openface/training/data.lua:36: in function 'addjob'
        /home/bamos/openface/training/train.lua:48: in function 'train'
        ./main.lua:35: in main chunk
        [C]: in function 'dofile'
        ...amos/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00406670
```

**Comment by melgor** [comment]
hi, any clue how to resolve it? I gave the same error.

**Comment by bamos** [comment]
Hi, yes, I just looked at this again and it's because I forgot to specify the `nn2` model for images sized 224 and the training was defaulting to the `nn4` model, which only supports inputs of 96x96. Because of this, the training code was passing in a tensor with 20 images, sized `20,3,224,224` and was receiving back a tensor sized `500,128` instead of `20,128`.

+ [ ] This error is very confusing and I'll add a check to prevent mis-matched sizes like this.

-Brandon.

**Comment by Prithviraj7** [comment]
What is the format of the training data in the folder ~/openface/data/casia-facescrub/dlib-affine-224-split/ ?

**Comment by bamos** [comment]
Hi @Prithviraj7 - `train` and `val` subdirectories containing directories of people with images.
This can be produced with steps 1 and 2 of 'Training new models' in the README:

---

1. Create a directory for your raw images so that images from different
   people are in different subdirectories. The names of the labels or
   images do not matter, and each person can have a different amount of images.
   The images should be formatted as `jpg` or `png` and have
   a lowercase extension.

   ```
   $ tree data/mydataset/raw
   person-1
   ├── image-1.jpg
   ├── image-2.png
   ...
   └── image-p.png

   ...

   person-m
   ├── image-1.png
   ├── image-2.jpg
   ...
   └── image-q.png
   ```


2. Preprocess the raw images, change `8` to however many
   separate processes you want to run:
   `for N in {1..8}; do ./util/align-dlib.py <path-to-raw-data> align affine <path-to-aligned-data> --size 96 &; done`.
   Prune out directories with less than N (I use 10) images
   per class with `./util/prune-dataset.py <path-to-aligned-data> --numImagesThreshold <N>` and
   then split the dataset into `train` and `val` subdirectories
   with `./util/create-train-val-split.py <path-to-aligned-data> <validation-ratio>`.

**Comment by bamos** [comment]
I've added an `imgDim` attribute to model definitions. The training code will error if it's not set or if the command-line option is incorrectly set. I decided to keep the command-line option so the user is aware of this resizing and in case a user's loading from a pre-trained model that they don't have a model definition for.

-Brandon.
